poly-commitment/src/evaluation_proof.rs

```diff
@@ -0,0 +1,306 @@
+use crate::commitment::*;
+use crate::srs::SRS;
+use ark_ec::{msm::VariableBaseMSM, AffineCurve, ProjectiveCurve};
+use ark_ff::{Field, One, PrimeField, UniformRand, Zero};
+use ark_poly::univariate::DensePolynomial;
+use oracle::{sponge::ScalarChallenge, FqSponge};
+use rand_core::{CryptoRng, RngCore};
+use rayon::prelude::*;
+use std::iter::Iterator;
+
+type Fr<G> = <G as AffineCurve>::ScalarField;
+type Fq<G> = <G as AffineCurve>::BaseField;
+
+/// This function opens polynomial commitments in batch
+///     plnms: batch of polynomials to open commitments for with, optionally, max degrees
+///     elm: evaluation point vector to open the commitments at
+///     polyscale: polynomial scaling factor for opening commitments in batch
+///     evalscale: eval scaling factor for opening commitments in batch
+///     oracle_params: parameters for the random oracle argument
+///     RETURN: commitment opening proof
+#[allow(clippy::too_many_arguments)]
+#[allow(clippy::type_complexity)]
+#[allow(clippy::many_single_char_names)]
+impl<G: CommitmentCurve> SRS<G> {
```

Comment:
I think the comment and annotations above apply to the function below, and not to this whole `impl` block. Can you move this `impl` line up above them?

Reply:
> I think the comment and annotations above apply to the function below, and not to this whole `impl` block. Can you move this `impl` line up above them?

Yes, of course.

Reply:
Thank you!